torch/distributed/device_mesh.py

```diff
@@ -785,10 +750,30 @@ def __getitem__(
                 # yet and need a follow-up,
                 # TODO: compiler + device_mesh slicing.
                 with torch._subclasses.fake_tensor.unset_fake_temporarily():
-                    submesh = _mesh_resources.create_sub_mesh(
-                        self, mesh_dim_names, slice_mesh_dims
+                    # When users sliced dim_names outside from current mesh, we will check whether
+                    # there is layout overlap. Eventually we will just directly throw error here because
```

Comment:
what does layout overlap mean? the idea is, if you tried to slice ('dp', and 'dp_tp') this would error?
can't you express this nicely in terms of layout algebra too?

Reply:
> if you tried to slice ('dp', and 'dp_tp') this would error?

correct that is the purpose.

> can't you express this nicely in terms of layout algebra too?

This and your question in L600 are great points, I thought about it but I didn't find anything we can leverage at this moment. Originally I just want to quickly finish the refactor and check its feasibility and we do see some more complicated picture here now. Let me look further into CuTe and see how we can also leverage layout algebra for this and L600.